src/service/viewer-impl.js

```diff
@@ -463,7 +477,50 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    const state = this.visibilityState_;
```

Comment:
The problems is I would have to keep track of what the viewer says the vis state is independently of what I store in this property.

Imagine the viewer sends down `PAUSED` and the tab is visible. In this case, we set `visibilityState_ = PAUSED`. Then, the user changes tabs so we update our vis state to `visibilityState_ = HIDDEN`. Then we change tabs back, so we should go back to `PAUSED`. But we don't store that information anywhere other than `visibilityState_` which is now `HIDDEN`.